### PR TITLE
Add Yaml Description

### PR DESCRIPTION
**Pull request by axxroytovu**
Adds the optional `"yaml_description"` item to the `Categories.json` file, which allows the user to add a line of description text to the yaml options.

Example syntax:
```json
{
    "Example Yaml-option category": {
        "yaml_option": ["DLC_enabled"],
        "yaml_description": {"DLC_enabled": "Include items and locations that require purchasing the DLC"}
    }
}
```

Result in the player yaml:
```yaml
  DLC_enabled:
    # Include items and locations that require purchasing the DLC
    'false': 0
    'true': 50
```

### CODE REVIEW

File: schemas/Manual.categories.schema.json [hunk 1]
```diff
@@ -30,6 +30,10 @@
                             "minItems": 1,
                             "uniqueItems": true
                         },
+                        "yaml_description": {
+                            "description": "(Optional) Dictionary of option-description pairs to add descriptor text to the yaml options defined in 'yaml_option'",
+                            "type": "object"
+                        }
```
Comment (nicopop) [inline schemas/Manual.categories.schema.json:36]: personally I would define the object using paternproperties but this should work as is

**Review by nicopop — APPROVED** [review]
this feels more like something that should be done in a dedicated option section/file like the wip #54
In the short term, yes this works but im not 100% happy with it but I leave the decisions to fuzzy

File: schemas/Manual.categories.schema.json (hunk 1 above)
Comment (axxroytovu) [reply schemas/Manual.categories.schema.json:36]: Yeah I've never used schemas so I did the bare minimum here. Feel free to suggest whatever makes sense to change here.

How do patternproperties work?

**Comment by axxroytovu** [comment]
> this feels more like something that should be done in a dedicated option section/file like the wip #54 In the short term, yes this works but im not 100% happy with it but I leave the decisions to fuzzy

I didn't know how far along #54 was so I wanted to push this fairly simple change in the short term.

File: schemas/Manual.categories.schema.json (hunk 1 above)
Comment (nicopop) [reply schemas/Manual.categories.schema.json:36]: Basicaly normaly with an object you can define its properties using `properties` but, when they are define by the user you can use a regex aka a pattern using `patternProperties` 
you could look at [items.schema.json](https://github.com/ManualForArchipelago/Manual/blob/main/schemas/Manual.items.schema.json#L77-L90) for a pretty good example for your situation

**Comment by silasary** [comment]
I do not like this syntax.

Putting a dict in each category is confusing, and the moment people start using the same option for multiple categories, they're going to be confused whether they need to duplicate the description.

If we're going to do this as a stopgap for #54, I'd want the dict in game.json, not in every category.

**Comment by FuzzyGamesOn** [comment]
> I do not like this syntax.
> 
> Putting a dict in each category is confusing, and the moment people start using the same option for multiple categories, they're going to be confused whether they need to duplicate the description.
> 
> If we're going to do this as a stopgap for #54, I'd want the dict in game.json, not in every category.

I agree with sila. Wherever it is, it has to be a single place to change it. And agree with both that the best single place would be the Options json when it happens.

game.json is a good fallback, but we know we'd be building something that we're going to deprecate probably this year. 

If we're really insistent on trying to get YAML option descriptions into world builders' hands, I'd almost prefer that we just build out a skeleton Options json that just takes an option name and lets you specify an option description, and nothing else. Then Nico's WIP PR can add types and other stuff later. That way, the YAML option description ends up in the file it's going to be in. We can explain away that Options in general are a WIP and that this would be a precursor to more functionality. (And, obviously, the format for this skeleton Options json should follow what Nico is establishing in their WIP PR).

**Comment by axxroytovu** [comment]
All good, I'll wait until Options.json is ready then.

**Pull request closed without merging**